File: python/Lec 2 Build-up Logical Thinking (Patterns)/pattern_02.py
#   Pattern:
#   *
#   **
#   ***
#   ****
#   *****

#----------------------------------
# BRUTE FORCE APPROACH
#----------------------------------
n = 5
for i in range(1, n + 1):
    for j in range(1, i + 1):
        print("*", end="")
    print()


#----------------------------------
# BETTER APPROACH (string repeat)
#----------------------------------
n = 5
for i in range(1, n + 1):
    print("*" * i)


#----------------------------------
# BEST APPROACH (Single loop building string)
#----------------------------------
n = 5
s = ""
for i in range(1, n + 1):
    s += "*"
    print(s)

File: cpp/Lec 2 Build-up Logical Thinking (Patterns)/pattern_02.cpp
//  Pattern:
//  *
//  **
//  ***
//  ****
//  *****

//----------------------------------
// BRUTE FORCE APPROACH
//----------------------------------
#include <iostream>
using namespace std;
int main()
{
    int n = 5;
    for (int i = 1; i <= n; i++)
    {
        for (int j = 1; j <= i; j++)
        {
            cout << "*";
        }
        cout << endl;
    }
    return 0;
}

//----------------------------------
// BETTER APPROACH (Using string constructor)
//----------------------------------
#include <iostream>
#include <string>
using namespace std;
int main2()
{
    int n = 5;
    for (int i = 1; i <= n; i++)
    {
        cout << string(i, '*') << endl;
    }
    return 0;
}

//----------------------------------
// BEST APPROACH (Single loop)
//----------------------------------
#include <iostream>
using namespace std;
int main3()
{
    int n = 5;
    int count = 1;
    string s = "";
    for (int i = 1; i <= n; i++)
    {
        s += "*";
        cout << s << endl;
    }
    return 0;
}